packages/design-system/src/components/tree/tree.tsx

```diff
@@ -442,6 +456,10 @@ const useKeyboardNavigation = <Data extends { id: string }>({
     rootRef,
     handleKeyDown,
     handleClick(event: React.MouseEvent<Element>) {
+      if (editingItemId) {
```

Comment:
You can use `event.detail === 1` and `event.detail === 2` to distinct click and double click.

Reply:
ah, right. It's to prevent click while editing.
Can be bypassed to TreeItemBody. You can use preventDefault() in your handler and for example check event.defaultPrevented to avoid calling selection

Reply:
Did that, didn't help much in some edge cases 😅. Instead it have new issues. `Space` doesn't add space  in `contenteditable` with `preventDefault` As it's prevent default behaviour 😬.  Let me change with `Input` and see if that solves most of these cases 👍